lib/src/native/realm_core.dart

```diff
@@ -1413,36 +1429,37 @@ class _RealmCore {
       _realmLib.realm_app_config_set_platform_version(handle._pointer, Platform.operatingSystemVersion.toCharPtr(arena));
 
       _realmLib.realm_app_config_set_sdk(handle._pointer, 'Dart'.toCharPtr(arena));
-      _realmLib.realm_app_config_set_sdk_version(handle._pointer, ''.toCharPtr(arena));
+      _realmLib.realm_app_config_set_sdk_version(handle._pointer, libraryVersion.toCharPtr(arena));
 
+      //TODO: set cpu architecture when Core merges https://github.com/realm/realm-core/pull/6256#pullrequestreview-1275624284
       _realmLib.realm_app_config_set_cpu_arch(handle._pointer, ''.toCharPtr(arena));
 
-      _realmLib.realm_app_config_set_device_name(handle._pointer, ''.toCharPtr(arena));
-      _realmLib.realm_app_config_set_device_version(handle._pointer, ''.toCharPtr(arena));
+      final deviceName = getDeviceName();
+      _realmLib.realm_app_config_set_device_name(handle._pointer, deviceName.toCharPtr(arena));
+      
+      final deviceVersion = getDeviceVersion();
+      _realmLib.realm_app_config_set_device_version(handle._pointer, deviceVersion.toCharPtr(arena));
 
       _realmLib.realm_app_config_set_framework_name(handle._pointer, (isFlutterPlatform ? 'Flutter' : 'Dart VM').toCharPtr(arena));
       _realmLib.realm_app_config_set_framework_version(handle._pointer, Platform.version.toCharPtr(arena));
 
-      _realmLib.realm_app_config_set_local_app_name(handle._pointer, ''.toCharPtr(arena));
-      _realmLib.realm_app_config_set_local_app_version(handle._pointer, ''.toCharPtr(arena));
-
       _realmLib.realm_app_config_set_base_url(handle._pointer, configuration.baseUrl.toString().toCharPtr(arena));
 
       _realmLib.realm_app_config_set_default_request_timeout(handle._pointer, configuration.defaultRequestTimeout.inMilliseconds);
 
       if (configuration.localAppName != null) {
         _realmLib.realm_app_config_set_local_app_name(handle._pointer, configuration.localAppName!.toCharPtr(arena));
       }
+      else {
+        _realmLib.realm_app_config_set_local_app_name(handle._pointer, ''.toCharPtr(arena));
+      }
 
       if (configuration.localAppVersion != null) {
         _realmLib.realm_app_config_set_local_app_version(handle._pointer, configuration.localAppVersion!.toCharPtr(arena));
+      } else {
+        _realmLib.realm_app_config_set_local_app_version(handle._pointer, ''.toCharPtr(arena));
```

Comment:
Should we use https://pub.dev/packages/package_info_plus

Reply:
Same. Do we need to call this method for setting nothing?

Reply:
as above